mismi-s3/src/Mismi/S3/Commands.hs

```diff
@@ -269,7 +283,7 @@ downloadWithMode mode a f = do
   unlessM (exists a) . fail $ "Can not download when the source does not exist [" <> (T.unpack $ addressToText a) <> "]."
   liftIO $ createDirectoryIfMissing True (dropFileName f)
   r <- getObject' a
-  r' <- maybe (fail "shit") pure r
+  r' <- maybe (fail $ "Unable to retrieve content for source [" <> (T.unpack $ addressToText a) <> "].") pure r
   liftIO . runResourceT . ($$+- sinkFile f) $ r' ^. gorsBody ^. to bodyResponse
```

Comment:
Yeah looks like `downloadWIthMode` might have been missed in the cleanup, still has a number of fails